app/pods/components/documents/document-card/template.hbs

```diff
@@ -116,55 +129,61 @@
+              {{moment-format this.piece.created "HH:mm"}}
+              {{#if this.piece.accessLevelLastModified}}
                 <div>
                   {{t "access-level-last-modified"}}
-                  {{moment-format this.lastPiece.accessLevelLastModified}}
+                  {{moment-format this.piece.accessLevelLastModified}}
                   {{t "at"}}
-                  {{moment-format this.lastPiece.accessLevelLastModified "HH:mm"}}
+                  {{moment-format this.piece.accessLevelLastModified "HH:mm"}}
                 </div>
               {{/if}}
             </div>
           </div>
         </div>
       </div>
     </div>
-    {{#if this.reverseSortedPieces.length}}
-      <div class="vlc-document-card__more">
-        <div class="js-vl-accordion {{if this.isExpandedVersionHistory "js-vl-accordion--open"}}">
-          <button
-            class="vl-toggle vl-link vl-link--bold"
-            type="button"
-            data-test-show-pieces-history
-            {{on "click" this.toggleVersionHistory}}
-          >
-            <i class="vl-link__icon vl-link__icon--before vl-toggle__icon ki-chevron-right"
-            ></i>
-            {{t "show-pieces"}}
-          </button>
-          {{#if this.isExpandedVersionHistory}}
-            <div class="vl-accordion__content js-vl-accordion__content">
-              <div class="vl-accordion__panel">
-                {{#each this.reverseSortedPieces as |piece|}}
-                  <WebComponents::VlDocument
-                    @onDeletePieceFromContainer={{@didDeletePiece}}
-                    @piece={{piece}} />
-                {{/each}}
-              </div>
-            </div>
+    <div class="vlc-document-card__more">
+      <div class="js-vl-accordion {{if this.isExpandedVersionHistory "js-vl-accordion--open"}}">
+        <button
+          class="vl-toggle vl-link vl-link--bold"
+          type="button"
+          disabled={{this.loadPieceRelatedData.isRunning}}
+          data-test-show-pieces-history
+          {{on "click" (if this.isExpandedVersionHistory this.collapseVersionHistory (perform this.expandVersionHistory))}}
+        >
+          <i class="vl-link__icon vl-link__icon--before vl-toggle__icon ki-chevron-{{if this.isExpandedVersionHistory "down" "right"}}"
+          ></i>
+          {{t "show-pieces"}}
+          {{#if this.expandVersionHistory.isRunning}}
+            <WebComponents::AuLoader/>
           {{/if}}
-        </div>
+        </button>
+        {{#if this.isExpandedVersionHistory}}
+          <div class="vl-accordion__content js-vl-accordion__content">
+            <div class="vl-accordion__panel">
+              {{#each this.visiblePieces as |piece|}}
+                {{!-- TODO: also  disable delete when piece linked to other agenda-items while maintaining DDAU --}}
+                <WebComponents::VlDocument
+                  @piece={{piece}}
+                  @enableDelete={{eq piece @piece}}
```

Comment:
Zoals besproken, deze enable delete is mogelijk niet grondig genoeg.
Vanuit de developer kant hadden we vóór deze refactor een check:
Als een piece is gelinkt aan meer dan 1 agendaitem mag die niet verwijderd worden, omdat in sommige gevallen we dan de link verliezen naar de documentContainer op oudere agendas

Die check kwam van onze kant als noodzaak, en die is nu er terug uit.
Het is maar de vraag wat we daar willen doen. Die check is nooit gevraagd geweest door de klant.
Dat de delete knop nu terug is betekent dan dat de klant mogelijk de data in een kapotte state kan achterlaten.
Willen we die check terug of vertrouwen we erop dat de klant weet wat hij/zij aan het doen is ?

Reply:
Als we de check niet gaan terugzetten, moet de test in agendaitem-pieces.spec aangepast worden, en dan zal deze branch groen zijn in jenkins.

Reply:
Zie https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/699/commits/356234c78161f5af1e93ebd22b43012f23d6c8f0